frontend/src/metabase/components/OnClickOutsideWrapper.jsx

```diff
@@ -1,74 +1,55 @@
-/* eslint-disable react/prop-types */
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 import ReactDOM from "react-dom";
 
-import { KEYCODE_ESCAPE } from "metabase/lib/keyboard";
-
-// keep track of the order popovers were opened so we only close the last one when clicked outside
-const popoverStack = [];
+import {
+  RENDERED_POPOVERS,
+  removePopoverData,
+  shouldClosePopover,
+} from "metabase/hooks/use-sequenced-content-close-handler";
 
 export default class OnClickOutsideWrapper extends Component {
   static propTypes = {
+    children: PropTypes.node,
     handleDismissal: PropTypes.func.isRequired,
   };
 
-  static defaultProps = {
-    dismissOnClickOutside: true,
-    dismissOnEscape: true,
-  };
-
   componentDidMount() {
     // necessary to ignore click events that fire immediately, causing modals/popovers to close prematurely
     this._timeout = setTimeout(() => {
-      popoverStack.push(this);
+      const contentEl = ReactDOM.findDOMNode(this);
+
+      this.popoverData = {
+        contentEl,
+        close: () => this.props.handleDismissal(),
+      };
+
+      RENDERED_POPOVERS.push(this.popoverData);
 
       // HACK: set the z-index of the parent element to ensure it"s always on top
       // NOTE: this actually doesn"t seem to be working correctly for popovers since PopoverBody creates a stacking context
-      ReactDOM.findDOMNode(this).parentNode.style.zIndex =
-        popoverStack.length + 2; // HACK: add 2 to ensure it"s in front of main and nav elements
+      contentEl.parentNode.style.zIndex = RENDERED_POPOVERS.length + 2; // HACK: add 2 to ensure it"s in front of main and nav elements
 
-      if (this.props.dismissOnEscape) {
-        document.addEventListener("keydown", this._handleKeyPress, false);
-      }
-      if (this.props.dismissOnClickOutside) {
-        window.addEventListener("mousedown", this._handleClick, true);
-      }
+      document.addEventListener("keydown", this._handleEvent, false);
+      window.addEventListener("mousedown", this._handleEvent, true);
     }, 0);
   }
 
   componentWillUnmount() {
-    document.removeEventListener("keydown", this._handleKeyPress, false);
-    window.removeEventListener("mousedown", this._handleClick, true);
+    document.removeEventListener("keydown", this._handleEvent, false);
```

Comment:
We don't seem to use function names prepended by `_` anymore, do we?

If that is correct, maybe in the future we update the names in this file.

Reply:
True -- it'll hopefully be deleted entirely soon enough.